serrano/resources/query/__init__.py

```diff
@@ -0,0 +1,116 @@
+from django.conf.urls import patterns, url
+from django.views.decorators.cache import never_cache
+
+from avocado.models import DataQuery
+from serrano.resources import templates
+from serrano.resources.history import ObjectRevisionResource, \
+    ObjectRevisionsResource, RevisionsResource
+from serrano.resources.query.base import PublicQueriesResource, \
+    QueryResource, QueriesResource
+from serrano.resources.query.forks import QueryForksResource
+from serrano.resources.query.results import QueryResultsResource
+from serrano.resources.query.stats import QueryStatsResource
+
+single_resource = never_cache(QueryResource())
+active_resource = never_cache(QueriesResource())
+public_resource = never_cache(PublicQueriesResource())
+forks_resource = never_cache(QueryForksResource())
+stats_resource = never_cache(QueryStatsResource())
+results_resource = never_cache(QueryResultsResource())
+
+revisions_resource = never_cache(RevisionsResource(
+    object_model=DataQuery, object_model_template=templates.Query,
+    object_model_base_uri='serrano:queries'))
+revisions_for_object_resource = never_cache(ObjectRevisionsResource(
+    object_model=DataQuery, object_model_template=templates.Query,
+    object_model_base_uri='serrano:queries'))
+revision_for_object_resource = never_cache(ObjectRevisionResource(
+    object_model=DataQuery, object_model_template=templates.Query,
+    object_model_base_uri='serrano:queries'))
+
+
+# Resource endpoints
+urlpatterns = patterns(
+    '',
+
+    url(
+        r'^$',
+        active_resource,
+        name='active'
+    ),
+
+    # Endpoints for specific queries
+    url(
+        r'^public/$',
+        public_resource,
+        name='public'
+    ),
+
+    # Single queries
+    url(
+        r'^(?P<pk>\d+)/$',
+        single_resource,
+        name='single'
+    ),
+    url(
+        r'^session/$',
+        single_resource,
+        {'session': True},
+        name='session'
+    ),
+
+    # Endpoint for retrieving results of an existing query.
```

Comment:
Added session-based endpoints for query results in 3a0cfef8bf62e8040b9b751ce7c4f420d829e720